src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts

```diff
@@ -134,10 +134,21 @@ export class CourseExerciseRowComponent implements OnInit, OnDestroy {
                 }
                 return ParticipationStatus.QUIZ_FINISHED;
             }
-        } else if ((exercise.type === ExerciseType.MODELING || exercise.type === ExerciseType.TEXT) && this.hasParticipations(exercise)) {
+        } else if (
+            (exercise.type === ExerciseType.MODELING || exercise.type === ExerciseType.TEXT || exercise.type === ExerciseType.FILE_UPLOAD) &&
+            this.hasParticipations(exercise)
+        ) {
             const participation = exercise.studentParticipations[0];
-            if (participation.initializationState === InitializationState.INITIALIZED || participation.initializationState === InitializationState.FINISHED) {
-                return exercise.type === ExerciseType.MODELING ? ParticipationStatus.MODELING_EXERCISE : ParticipationStatus.TEXT_EXERCISE;
+            if (participation.initializationState === InitializationState.INITIALIZED) {
+                if (this.isExerciseInDueDate(exercise)) {
```

Comment:
3rd level of nesting, please find a better solution to structure the control flow.

E.g. here:
```
return this.isExerciseInDueDate(exercise) ? ParticipationStatus.EXERCISE_ACTIVE : ParticipationStatus.EXERCISE_MISSED;

Reply:
I completely restructured the `participationStatus()` method :-)